pkg/schedule/placement/rule_manager.go

```diff
@@ -492,30 +493,56 @@ func (m *RuleManager) TryCommitPatch(patch *RuleConfigPatch) error {
 }
 
 func (m *RuleManager) savePatch(p *ruleConfig) error {
```

Comment:
it could meet "duplicate key given in txn request", where etcd not allow this case.